### ISSUE TITLE
Add domains to vPool/StorageRouter combo

### ISSUE THREAD

**Issue by wimpers** [openvstorage/framework#1212]
Currently domains are set at an Storage Router level. We could bring it down to the vPool/storage router level on the vpool details page/ management actions (where we are showing the accelerated alba for the storage router)

**Comment by khenderick** [comment]
Can you elaborate why this would be required? I don't see much benefit because Domains somewhat represents a physical location, and I'm not sure it makes sense to allow a StorageRouter can be part of multiple Domains.

**Comment by wimpers** [comment]
Now we define domain and recovery domain on SC level so basically you are saying that all vDisks on that SC need a DTL or MDS in another domain/datacenter. You might want to be able to overrule this on a vPool level so: 
* Global vPool: domain = NY, Recovery domain NJ (for SC in NY and vice versa for the ones in NJ)
* Local vPool: domain: rack 1, Recovery domain rack 2

The behaviour is now semi do-able by overriding it per vDisk.

**Comment by wimpers** [comment]
>  I don't see much benefit because Domains somewhat represents a physical location, and I'm not sure it makes sense to allow a StorageRouter can be part of multiple Domains.

I would agree if  domains would be (fully connected) graph (with weighted edges) so in that case you can select a leaf: Rack1 (Earth->US->NY->DC1->Cage1->Rack1)

**Comment by khenderick** [comment]
Hmm, you use case does make a lot of sense, but I'm afraid that making multiple levels of domains "a domain NY" and a domain "rack 1" might make things a lot more confusing. Needs some more thought.

**Comment by wimpers** [comment]
Seems the basic use case is already covered:

The DTL also keeps the vPool in mind so if the vPool isn't extended to that node it will not run the DTL. If it can't find any nodes to run in the Recovery domain, it will faalback and try to find one in the primary domain.
